config/manifests/bases/external-dns-operator.clusterserviceversion.yaml

```diff
@@ -7,9 +7,15 @@ metadata:
     certified: "false"
     containerImage: quay.io/openshift/origin-external-dns-operator:latest
     createdAt: 2021/09/28
+    features.operators.openshift.io/disconnected: "true"
+    features.operators.openshift.io/fips-compliant: "true"
+    features.operators.openshift.io/proxy-aware: "true"
+    features.operators.openshift.io/tls-profiles: "false"
+    features.operators.openshift.io/token-auth-aws: "false"
+    features.operators.openshift.io/token-auth-azure: "false"
+    features.operators.openshift.io/token-auth-gcp: "false"
     olm.skipRange: <1.2.0
     operatorframework.io/suggested-namespace: external-dns-operator
-    operators.openshift.io/infrastructure-features: '["disconnected","proxy-aware"]'
```

Comment:
For the FIPS compliance, it's not 100% clear indeed. Hongan [says the ExtDNSOperator was tested on FIPS](https://github.com/openshift/external-dns-operator/pull/213#discussion_r1447255388) but similar to ALBO we don't have a CI which verifies this continuously. Would you prefer to switch it back to `false`?

Reply:
> For the FIPS compliance, it's not 100% clear indeed. Hongan [says the ExtDNSOperator was tested on FIPS](https://github.com/openshift/external-dns-operator/pull/213#discussion_r1447255388) but similar to ALBO we don't have a CI which verifies this continuously. Would you prefer to switch it back to `false`?

We had better ask the Portfolio Enablement team what is required to claim "fips-compliant" as a feature.  In any case, the commit message should explain why the change was made.  At the very least, something like, "We tested the operator on a FIPS-enabled cluster, and the operator passed its tests."  If there is a public reference for the requirements for claiming "fips-compliant", please cite it in the commit message as well.  

> The auth feature was not implemented for the ExternalDNS Operator so it should remain `false`.
> 
> The `tls-profiles` feature is described as follows:
> 
> ```
> features.operators.openshift.io/tls-profiles 	Whether the operator implements well-known tunables to modify the TLS cipher suite used by the operator and, if applicable, any of the workloads it manages (operands).
> ```

Is the expectation now that the manifest explicitly specify each feature that it *does not* support?  (I guess we will need to keep adding annotations every time a new one is defined!)  Would you mind adding a statement in the commit message that these features have not been implemented?

Reply:
[The downstream OLM docs](https://docs.engineering.redhat.com/display/CFC/Best_Practices) describes the FIPS compliance as follows:

> Whether the opperator accepts the FIPS-140 configuration of the underlying platform and works on nodes that are booted into FIPS mode. In this mode, the operator and any workloads it manages (operands) are solely calling the RHEL cryptographic library submitted for FIPS-140 validation.

Both [the operator](https://github.com/openshift/external-dns-operator/blob/main/Makefile#L137) and [operand](https://github.com/openshift/external-dns/blob/master/Makefile#L100) are built statically which is not FIPS compliant according to [Checking FIPS compliance slide](https://docs.google.com/presentation/d/1o3IowxHX6BsnxGkIInaQ0lBgnn_K5Ex8jxwCYCeNsqs/edit#slide=id.g276351f5efb_2_1).

I reverted the `fips-compliant` annotation to `false`.

> Is the expectation now that the manifest explicitly specify each feature that it does not support? (I guess we will need to keep adding annotations every time a new one is defined!) 

I think so, yes. A quote from [the downstream OLM docs](https://docs.engineering.redhat.com/display/CFC/Best_Practices) about the purpose of this new set of annotations:

> This is a change from an earlier convention [up until OpenShift 4.13](https://docs.openshift.com/container-platform/4.13/operators/operator_sdk/osdk-generating-csvs.html#osdk-csv-manual-annotations_osdk-generating-csvs) where the annotation operators.openshift.io/infrastructure-features was used as a mere list. The old annotations weren't enforced in the pipeline because of their format providing no way to differentiate between deliberate absence of a value from human obliviousness.



> Would you mind adding a statement in the commit message that these features have not been implemented?

Added.